### ISSUE TITLE
Monitor queue size

### ISSUE THREAD

**Issue by henare** [openaustralia/morph#989]
We've got a big queue again (280 items) and I didn't know about it. Can we do this in New Relic easily?

**Comment by auxesis** [comment]
@henare which queue are we talking about? I assume a Sidekiq one? 

If so, is it addressed by [the new graphs](http://morph-monitor.fractio.nl/dashboard/db/morph?from=now-2d&to=now)?

**Comment by henare** [comment]
Yes, the scraper queue. From my comment about it also sounds like I meant alerting too. Maybe we don't need that?

**Comment by auxesis** [comment]
Is anyone going to respond to the alert? If not, don't set one up – it'll just be noise.

**Comment by auxesis** [comment]
> _If so, is it addressed by the new graphs?_

For future readers, this is a screenshot of a graph I'm talking about:

![image](https://cloud.githubusercontent.com/assets/12306/22495355/bad40ff2-e893-11e6-8820-086725de7761.png)

**Comment by henare** [comment]
>Is anyone going to respond to the alert? If not, don't set one up – it'll just be noise.

OP has the rationale:

>We've got a big queue again (280 items) and I didn't know about it.

But I guess we get a Honeybadger notification when there's no more slots for the first time anyway now?

**Comment by auxesis** [comment]
Yep. 

Choose one way of being notified, and use that.

**Comment by equivalentideas** [comment]
> If so, is it addressed by the new graphs?

@auxesis I don't think so, because it's actually retries that we're most interested in the current configuration. When Henare describes the "We've got a big queue again (280 items)" that would be 280 jobs in the retries queue.

I think it would be most useful to have one chart for scheduled jobs so we can see if that's not draining properly. And one that shows retries ( and maybe busy in there as well?) so we can see if there's a big backlog.

**Comment by auxesis** [comment]
If recovery is not time critical, then an email is an OK place for the alert to go and die.

**Comment by auxesis** [comment]
> I think it would be most useful to have one chart for scheduled jobs so we can see if that's not draining properly. And one that shows retries ( and maybe busy in there as well?) so we can see if there's a big backlog.

Gotcha @equivalentideas. I think the [collectd-sidekiq-plugin](https://github.com/auxesis/collectd-sidekiq-plugin) needs some work so it displays stats from the retry queues properly.

**Comment by auxesis** [comment]
>  I think the collectd-sidekiq-plugin needs some work so it displays stats from the retry queues properly.

Per @equivalentideas's feedback, I've added support in [collectd-sidekiq-plugin](https://github.com/auxesis/collectd-sidekiq-plugin) to [collect retries, scheduled, dead, and number of processes](https://github.com/auxesis/collectd-sidekiq-plugin/releases/tag/0.3.0). 

This is [deployed onto the Morph graphs](http://morph-monitor.fractio.nl/dashboard/db/morph?from=now-3h&to=now): 

![image](https://cloud.githubusercontent.com/assets/12306/22547992/6ccf2f5c-e998-11e6-9658-4ad873b41314.png)

![image](https://cloud.githubusercontent.com/assets/12306/22548002/75993bd2-e998-11e6-9985-83a3569e6e57.png)

I've also added a panel that calculates the current percentage of jobs that are failed vs succeeded:

![image](https://cloud.githubusercontent.com/assets/12306/22548020/8de450aa-e998-11e6-88e3-42b7137ddc15.png)

The colour thresholding is set to go orange on > 50, and red on > 80.

**Comment by auxesis** [comment]
Here's the _retries, dead, scheduled_ graph with a log(10) scale on the _y_ axis:

![image](https://cloud.githubusercontent.com/assets/12306/22565846/c6726b1a-e9dd-11e6-84e3-1f8152c73c2a.png)

This makes it a bit easier to see changes to retries at the same time as the number of scheduled jobs.

**Comment by equivalentideas** [comment]
> Here's the retries, dead, scheduled graph with a log(10) scale on the y axis:

![screen shot 2017-02-06 at 4 24 36 pm](https://cloud.githubusercontent.com/assets/1239550/22635628/f9dd9d46-ec88-11e6-8f66-42ed9a172507.png)

This is suiting my use well.

I recon it should span the whole way across to match the load chart above.

It's much more important than the 'processed/failed' chart that it's currently next to, because that chart doesn't really move in a noticeable way on production (there's over 2 million jobs so you can't see slight changes in ratio).

**Comment by auxesis** [comment]
> I recon it should span the whole way across to match the load chart above.

@equivalentideas [how about this](http://morph-monitor.fractio.nl/dashboard/db/morph)?

![image](https://cloud.githubusercontent.com/assets/12306/22636526/0324b0ae-ec90-11e6-8f3b-b45235aac014.png)

**Comment by equivalentideas** [comment]
Nice 👍

**Comment by equivalentideas** [comment]
The charts are now really useful for seeing if something is going wrong, but we still don't have the notifying aspect in place.

@auxesis Whenever who've got time to add it, I think to close this issue we set an alert (like our disc space one) into our #morph slack channel, when the _retry queue goes over 50_ or _the enqueued queue goes over 50_ .
